packages/foundation/src/slots.test.tsx

```diff
@@ -2,7 +2,115 @@ import * as React from 'react';
 import * as renderer from 'react-test-renderer';
 import { mount } from 'enzyme';
 import { withSlots, createFactory, getSlots } from './slots';
-import { IProcessedSlotProps, ISlot, ISlotProp, ISlotPropRenderFunction, ISlotDefinition } from './ISlots';
+import { IHTMLElementSlot, IHTMLSlot } from './IHTMLSlots';
+import { ExtractProps, ExtractShorthand, IProcessedSlotProps, ISlot, ISlotDefinition, ISlotProp } from './ISlots';
+
+describe('typings', () => {
+  type ITestProps = { testProp?: number };
+  const TestProps = { testProp: 42 };
+  const TestOptions = { props: TestProps };
+  // TODO: this should be able to support returning nulls
+  // const TestRender =
+  //   (props: ITestProps, defaultComponent: React.ComponentType<ITestProps>): ReturnType<React.StatelessComponent> => null;
+  const TestRender = (props: ITestProps, defaultComponent: React.ComponentType<ITestProps>) => <div />;
+  const TestComponent: React.FunctionComponent<ITestProps> = () => null;
+
+  it('do not generate TS compile errors on getSlots usage', () => {
+    interface IComponentSlots {
+      testHtmlSlot?: IHTMLSlot;
+      testElementSlot?: IHTMLElementSlot<'button'>;
+      testSlot?: ISlotProp<ITestProps, string>;
+      testSlotNoShorthand?: ISlotProp<ITestProps>;
+    }
+
+    interface IComponentProps extends IComponentSlots {
+      someProp?: number;
```

Comment:
I don't see this being used anywhere.

Reply:
It wouldn't be, it's part of a typing test. Just want some key in this interface that's not part of `IComponentSlots`.